core/frontend/src/components/app/ImagePicker.vue

```diff
@@ -0,0 +1,255 @@
+          </v-alert>
+          <v-alert v-if="error" color="red lighten-2">
+            {{ error }}
+          </v-alert>
+        </v-row>
+        <v-row>
+          <v-col cols="12">
+            <v-card
+              class="drop-zone"
+              @dragover.prevent
+              @dragenter.prevent
+              @drop.prevent="onDrop"
+            >
+              <v-card-text>
+                Drag and drop an image file here to upload
+              </v-card-text>
+              <input
+                id="file-input"
+                ref="fileInput"
+                aria-label="File browser"
+                type="file"
+                accept="image/*"
+                style="display:none"
+                @change="onFileInputChange"
+              />
+              <v-btn color="primary" @click="onFilePickerClick">
+                Click to upload an image or
+              </v-btn>
+            </v-card>
+          </v-col>
+        </v-row>
+        <v-alert v-if="uploadError" color="red lighten-2">
+          {{ uploadError }}
+        </v-alert>
+      </v-card>
+    </v-dialog>
+  </div>
+</template>
+
+<script lang="ts">
+import axios from 'axios'
+import Vue from 'vue'
+
+import back_axios from '@/utils/api'
+
+import SpinningLogo from '../common/SpinningLogo.vue'
+
+interface FileEntry {
+  name: string;
+  type: string;
+  mtime: string;
+}
+
+export default Vue.extend({
+  name: 'ImagePicker',
+  components: {
+    SpinningLogo,
+  },
+  props: {
+    directory: {
+      type: String,
+      default: '/userdata/images/',
+      required: false,
+    },
+    image: {
+      type: String,
+      default: '/userdata/images/',
+      required: false,
+    },
+  },
+  data() {
+    return {
+      dialog: false,
+      selectedIndex: null as number | null,
+      images: [] as string[],
+      error: null as string | null,
+      loading: true,
+      uploadError: null as string | null,
+    }
+  },
+  computed: {
+  },
+  mounted() {
+    this.loadImages()
+  },
+  methods: {
+    selectImage(index: number) {
+      this.selectedIndex = index
+      this.$emit('image-selected', `${this.directory}/${this.images[index]}`)
+      this.dialog = false
+    },
+    openDialog() {
+      this.dialog = true
+    },
+    loadImages() {
+      back_axios({
+        method: 'get',
+        url: this.directory,
+      }).then((response) => {
+        this.images = response.data.filter(
+          (string: FileEntry) => string.type === 'file',
+        ).map((file: FileEntry) => file.name)
+        this.loading = false
+      }).catch((error) => {
+        this.error = error
+      })
+    },
+    async deleteImage(index: number) {
+      try {
+        await back_axios({
+          method: 'delete',
+          url: `upload/${this.directory}/${this.images[index]}`,
+        })
+        this.loadImages()
+      } catch (error) {
+        console.error('Error deleting file:', error)
```

Comment:
Should we raise a notification ?

Reply:
I'm not showing it on the alert component